wasm/bindings.cpp

```diff
@@ -0,0 +1,113 @@
+#include <emscripten/bind.h>
+
+using namespace emscripten;
+
+#include <manifold.h>
+
+using namespace manifold;
+
+void logMesh(Mesh& mesh) {
+    printf("# %d vertices:\n", (int)mesh.vertPos.size());
+    for(int i = 0; i < mesh.vertPos.size(); i++) {
+        printf("v %f %f %f\n",
+            mesh.vertPos[i].x,
+            mesh.vertPos[i].y,
+            mesh.vertPos[i].z
+        );
+    }
+    printf("# %d faces:\n", (int)mesh.triVerts.size());
+    for(int i = 0; i < mesh.triVerts.size(); i++) {
+        printf("f %d %d %d\n",
+            mesh.triVerts[i].x + 1,
+            mesh.triVerts[i].y + 1,
+            mesh.triVerts[i].z + 1
+        );
+    }
+}
+
+struct Mesh makeEmptyMesh() {
+  struct Mesh empty; return empty;
+}
+
+Manifold makeManifoldFromMesh(Mesh& mesh) {
+    //logMesh(mesh);
+
+    try {
+        Manifold manifold(mesh);
+        return manifold;
+    }
+    catch(const std::runtime_error& e) {
+        puts(e.what());
```

Comment:
This is interesting, and gets at some of the discussion around removing or changing exception handling: #141. Is there no way for C++ exceptions to automatically throw JS exceptions with WASM?

Reply:
@elalish it does throw all the way to js but the message you get is just some random number - so, with puts(e.what()) you can at least see what was actually wrong on c++ side